library/ssl_srv.c

```diff
@@ -1431,32 +1411,23 @@ static int ssl_parse_client_hello( mbedtls_ssl_context *ssl )
      */
     MBEDTLS_SSL_DEBUG_BUF( 3, "client hello, version", buf, 2 );
 
-    mbedtls_ssl_read_version( &ssl->major_ver, &ssl->minor_ver,
-                      ssl->conf->transport, buf );
-    ssl->session_negotiate->minor_ver = ssl->minor_ver;
+    tls_ver = mbedtls_ssl_read_version( buf, ssl->conf->transport );
+    ssl->tls_ver = tls_ver;
+    ssl->session_negotiate->tls_ver = tls_ver;
 
-    ssl->handshake->max_major_ver = ssl->major_ver;
-    ssl->handshake->max_minor_ver = ssl->minor_ver;
-
-    if( ssl->major_ver < ssl->conf->min_major_ver ||
-        ssl->minor_ver < ssl->conf->min_minor_ver )
+    if( tls_ver < ssl->conf->min_tls_ver )
     {
         MBEDTLS_SSL_DEBUG_MSG( 1, ( "client only supports ssl smaller than minimum"
-                            " [%d:%d] < [%d:%d]",
-                            ssl->major_ver, ssl->minor_ver,
-                            ssl->conf->min_major_ver, ssl->conf->min_minor_ver ) );
+                            " [0x%x] < [0x%x]",
+                            (unsigned) tls_ver,
+                            ssl->conf->min_tls_ver ) );
         mbedtls_ssl_send_alert_message( ssl, MBEDTLS_SSL_ALERT_LEVEL_FATAL,
                                      MBEDTLS_SSL_ALERT_MSG_PROTOCOL_VERSION );
         return( MBEDTLS_ERR_SSL_BAD_PROTOCOL_VERSION );
     }
 
-    if( ssl->major_ver > ssl->conf->max_major_ver )
-    {
-        ssl->major_ver = ssl->conf->max_major_ver;
-        ssl->minor_ver = ssl->conf->max_minor_ver;
-    }
-    else if( ssl->minor_ver > ssl->conf->max_minor_ver )
-        ssl->minor_ver = ssl->conf->max_minor_ver;
+    if( tls_ver > ssl->conf->max_tls_ver )
+        ssl->tls_ver = ssl->conf->max_tls_ver;
```

Comment:
If `ssl->tls_ver` is set to `ssl->conf->max_tls_ver`, shouldn't this also be done to `ssl->session_negotiate->tls_ver` ?  (See a few lines above where both are set.)